src/fiatconnect/saga.ts

```diff
@@ -277,14 +280,60 @@ export function* handleSelectFiatConnectQuote({
   payload: params,
 }: ReturnType<typeof selectFiatConnectQuote>) {
   const { quote } = params
+
   try {
+    // If KYC is required for the quote, check that the user has correct KYC on file
+    // with the quote's provider
+    let getKycStatusResponse: GetKycStatusResponse
+    const kycSchema = quote.getKycSchema()
+    if (kycSchema) {
+      getKycStatusResponse = yield call(getKycStatus, {
+        providerInfo: quote.quote.provider,
+        kycSchemas: [kycSchema],
+      })
+      const fiatConnectKycStatus = getKycStatusResponse.kycStatus[kycSchema]
+      switch (fiatConnectKycStatus) {
+        case FiatConnectKycStatus.KycNotCreated:
+          if (getKycStatusResponse.persona === PersonaKycStatus.Approved) {
+            // If user has Persona KYC on file, just submit it and continue to account management
+            yield call(postKyc, {
+              providerInfo: quote.quote.provider,
+              kycSchema,
+            })
+            break
+          } else {
+            // If no Persona KYC on file, navigate to Persona
+            navigate(Screens.Persona, {
+              kycStatus: getKycStatusResponse.persona,
+            })
+            yield put(selectFiatConnectQuoteCompleted())
+            return
+          }
+        // If approved or pending, continue as normal and handle account management
+        case FiatConnectKycStatus.KycApproved:
+        case FiatConnectKycStatus.KycPending:
+          break
+        // If denied or expired, skip account management and
+        // navigate to the KYC status screen
+        case FiatConnectKycStatus.KycDenied:
+        case FiatConnectKycStatus.KycExpired:
+          navigate(Screens.KycStatusScreen)
```

Comment:
This screen should eventually take flow, quote, and kyc status as props right?

Reply:
Yep, that's probably the case.